### ISSUE TITLE
[Question] appWithI18n should accept JSX.Element?

### ISSUE THREAD

**Issue by EricKwan2014** [vinissimus/next-translate#396]
![image](https://user-images.githubusercontent.com/8048626/101719825-19ce4a80-3adf-11eb-8812-714de996389a.png)

![image](https://user-images.githubusercontent.com/8048626/101719754-eab7d900-3ade-11eb-9cd2-84212c60c481.png)

**Comment by aralroca** [comment]
@EricKwan2014 How is your `MyApp` defined? It is likely that the types need to be improved. Feel free to do a PR yourself by adapting the types.

**Comment by EricKwan2014** [comment]
@aralroca 
First of all, thanks for the effort!
I am using the similar coding with https://github.com/vercel/next.js/blob/canary/examples/with-next-seo/pages/_app.js
So that it default returned the JSX.Element haha
Maybe I can change the scalar type in my project
But I am not sure is it a good idea to provide the flexibility in your lib or let user to strict the type in _app level 😕

**Comment by aralroca** [comment]
@EricKwan2014 You can define _app.js in all the possible ways that Next.js lets you do it. Perhaps it would be interesting to update the types to support more options.

**Comment by aralroca** [comment]
@EricKwan2014 it's already solved in **[1.0.1-canary.4](https://github.com/vinissimus/next-translate/releases/tag/1.0.1-canary.4) prerelease** 😊 

Can you confirm that it works for you? Thanks 🙏

**Comment by slevy85** [comment]
Hi, 
I am having the same issue using the version 1.0.7, can you help me with it ? I am not sure if it is a bug if I am doint it wrong, I am following the typings described in next documentation https://nextjs.org/docs/basic-features/typescript#custom-app.

This is my code :  
```
import { AppProps } from 'next/app'
import appWithI18n from 'next-translate/appWithI18n'

function MyApp({ Component, pageProps }: AppProps) {/**/}

export default appWithI18n(MyApp, {/**/})
```

And the typescript error is 
```
Argument of type '({ Component, pageProps }: AppProps<{}>) => Element' is not assignable to parameter of type 'NextComponentType<PartialNextContext, {}, {}>'.
  Type '({ Component, pageProps }: AppProps<{}>) => Element' is not assignable to type 'FunctionComponent<{}> & { getInitialProps?(context: PartialNextContext): {} | Promise<{}>; }'.
    Type '({ Component, pageProps }: AppProps<{}>) => Element' is not assignable to type 'FunctionComponent<{}>'.
      Types of parameters '__0' and 'props' are incompatible.
        Type '{ children?: ReactNode; }' is not assignable to type 'AppProps<{}>'.
          Property 'pageProps' is missing in type '{ children?: ReactNode; }' but required in type 'AppInitialProps'
```
